src/datafile.rs

```diff
@@ -100,9 +100,12 @@ impl DataFile {
             key: key.to_vec(),
             value: value.to_vec(),
         };
+        use std::io::Write as _;
 
         let offset = self.file.seek(SeekFrom::Current(0))?;
-        bincode::serialize_into(&mut *self.file, &entry)?;
+        // serialize_into is vastly slower than serializing to avec then doing 1 big write
+        let encoded: Vec<u8> = bincode::serialize(&entry)?;
+        self.file.write_all(&encoded)?;
```

Comment:
after getting the benchmarks to run, `serialize_into` is vastly slower than this. I don't know why.
prior to this it was using write, which didn't check how many bytes were written (could've been less than the vec). `write_all` ensures all bytes get written (write in a loop basically)

with `serialize_into`
```
test reads_16_k  ... bench:      82,936 ns/iter (+/- 10,547)
test reads_2_k   ... bench:      38,883 ns/iter (+/- 5,177)
test reads_32_k  ... bench:     135,088 ns/iter (+/- 7,206)
test reads_4_k   ... bench:      45,400 ns/iter (+/- 5,582)
test reads_8_k   ... bench:      57,978 ns/iter (+/- 3,574)
test writes_16_k ... bench:  65,971,520 ns/iter (+/- 2,540,212)
test writes_2_k  ... bench:   8,180,700 ns/iter (+/- 866,488)
test writes_32_k ... bench: 132,051,020 ns/iter (+/- 4,831,216)
test writes_4_k  ... bench:  16,665,630 ns/iter (+/- 1,583,663)
test writes_8_k  ... bench:  33,111,100 ns/iter (+/- 3,072,559)
```

with serialize to a vec and then a `write_all`
```
test reads_16_k  ... bench:     103,171 ns/iter (+/- 6,223)
test reads_2_k   ... bench:      45,552 ns/iter (+/- 4,165)
test reads_32_k  ... bench:     171,250 ns/iter (+/- 13,510)
test reads_4_k   ... bench:      54,255 ns/iter (+/- 10,917)
test reads_8_k   ... bench:      70,724 ns/iter (+/- 12,245)
test writes_16_k ... bench:     325,729 ns/iter (+/- 47,404)
test writes_2_k  ... bench:      50,829 ns/iter (+/- 15,355)
test writes_32_k ... bench:     621,725 ns/iter (+/- 113,400)
test writes_4_k  ... bench:     101,420 ns/iter (+/- 28,109)
test writes_8_k  ... bench:     176,533 ns/iter (+/- 42,273)
```

ignore the random deviance on the reads, my cpu was semi-busy.